### ISSUE TITLE
Papirus-icon-theme LibreOffice Calc wrong icon

### ISSUE THREAD

**Issue by sund3RRR** [NixOS/nixpkgs#231862]
### Describe the bug
After installing Papirus-icon-theme i have noticed, that LibreOffice Calc has wrong icon (like calculator app).

### Steps To Reproduce
1. Install LibreOffice
2. Install Papirus-icon-theme

![image](https://github.com/NixOS/nixpkgs/assets/73298492/cf7f373d-6364-46ab-ba9d-1aee1ccbd3a3)

**Comment by luis-caldas** [comment]
https://github.com/NixOS/nixpkgs/blob/12f5b0f81f3d1b74f09bc179307eb38a0dd8b470/pkgs/applications/office/libreoffice/default.nix#L384

I looked over a couple of icon themes and most of them have their references with the prefix `libreoffice-...`

**Comment by 7c6f434c** [comment]
Hmm, if I unpack the upstream tarball, it looks like the current upstream `Icon=` is actually `calc`?

(Note that I have zero clue about the `.desktop` entries for Libreoffice as I only ever launch it from command line)

**Comment by luis-caldas** [comment]
@7c6f434c you're right there, the upstream does indeed use `calc` for its icon, but the compiled `.rpm` & `.dev` all use `libreoffice{version}-calc`, which could be the ones that the icon themes are basing themselves on

I also had a look at a couple of themes

||libreoffice-calc|calc|
|-|-|-|
|papirus|✓||
|flat-remix|✓||
|luv|✓||
|la-capitaine|✓||
|numix|✓||

None of them related to the program as `calc` only as `libreoffice-calc` or `libreoffice{version}-calc`

**Comment by 7c6f434c** [comment]
So basically the sed expression should be _in the other direction_ now? I am fine merging it trusting the submitter's word that the submitted fix indeed works, of course.

I am even fine if the main expressions stays as it is with a TODO comment to drop the sed during the next bump, and it's just the wrapper that does the necessary magic (quick but annoying churnable things going to the wrapper reduces the debugging pain…)

**Comment by Aleksanaa** [comment]
Should be fixed.
